modules/apps/portal/portal-tika/src/main/java/com/liferay/portal/tika/internal/util/TikaConfigUtil.java

```diff
@@ -14,27 +14,69 @@
 
 package com.liferay.portal.tika.internal.util;
 
+import com.liferay.petra.io.StreamUtil;
+import com.liferay.petra.io.unsync.UnsyncByteArrayInputStream;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.exception.SystemException;
+import com.liferay.portal.kernel.util.PortalClassLoaderUtil;
+import com.liferay.portal.tika.internal.configuration.TikaConfiguration;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.util.Map;
+
 import org.apache.tika.config.TikaConfig;
 
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
 /**
  * @author Shuyang Zhou
+ * @author Jorge Díaz
  */
+@Component(
+	configurationPid = "com.liferay.portal.tika.internal.configuration.TikaConfiguration",
+	service = {}
+)
 public class TikaConfigUtil {
 
 	public static TikaConfig getTikaConfig() {
 		return _tikaConfig;
 	}
 
-	private static final TikaConfig _tikaConfig;
+	public static String getTikaConfigXml() throws IOException {
```

Comment:
@jorgediaz-lr The method name and configuration property name is confusing. This method "getTikaConfigXml" is actually getting tick config content, while the property name is also "tikaConfigXml" which means xml config file path.

Reply:
Hi @tinatian 

I have changed the **TikaConfigHelper.getTikaConfigXml()** method with a new one called **TikaConfigHelper.getTikaConfigInputStream()** that returns the InputStream, this avoid name confusion and it simplifies the code that calls this method, for more information see new commit https://github.com/liferay-core-infra/liferay-portal/commit/f7d5f0a7031ef225720719a3257a2ab7705755e3 and its usage in "TextExtractorImpl" in new commit https://github.com/liferay-core-infra/liferay-portal/commit/560f0b87d543cd7a355fb4aae645720d88e632ad

Let me know if you have any question.